example/classifier/odp_classifier.c

```diff
@@ -302,7 +302,7 @@ static odp_pktio_t create_pktio(const char *dev, odp_pool_t pool)
 static int pktio_receive_thread(void *arg)
 {
 	int thr;
-	odp_pktout_queue_t pktout;
+	odp_pktout_queue_t pktout[1] = {0};
```

Comment:
This is not very nice.

This is almost the same as if you simply added {0} initializer without making the variable an array and without making all the other changes that the change of the type requires.

The problem is that neither code is strictly correct since the ODP application code is not supposed to know what odp_pktout_queue_t actually is and you cannot have an initializer that would work with all possible types.

I wonder your use of the array is a trick to allow both structure types and simple integer and pointer types as odp_pktout_queue_t since the initializer would be valid for both cases (the braces of a structure initializer are optional when inside an array initializer). But such a trick is quite non-obvious and might confuse the reader. This is an example application after all.

I am not sure what the right workaround for the LTO warning is, but I feel that making the code too weird or not stricly ODP API compliant may not be the right way to go. The code could use memset, with a comment that it is only for suppressing bogus warnings, but maybe it would be better if ODP API provided ODP_PKTOUT_QUEUE_INVALID for this kind of initialization purposes.

Reply:
This has not been fixed in v2. If the initialization is really needed, I would suggest using memset and leaving the rest of the code alone. But is the initialization really needed instead of the underlying ODP implementation needing a fix?

Reply:
I had relook at our changes and fixed implementation rather with some minor changes. I am revoking this proposal. Thanks for your time reviewing these changes.